dispatcher.go

```diff
@@ -120,11 +157,20 @@ type multicastDispatcher[T any] struct {
 // AddEventHandler adds a new event handler method that is called whenever an event T is dispatched. A
 // unique HandlerID is returned that can be used to remove the handler.
 func (md *multicastDispatcher[T]) AddEventHandler(handler EventHandler[T]) HandlerID {
+	return md.AddFilteredEventHandler(handler, nil)
+}
+
+// AddFilteredEventHandler adds a new event handler method that is called whenever a dispatched event T
+// passes the provided condition. A unique HandlerID is returned that can be used to remove the handler.
+func (md *multicastDispatcher[T]) AddFilteredEventHandler(handler EventHandler[T], condition EventCondition[T]) HandlerID {
 	handlerID := HandlerID(uuid.NewString())
```

Comment:
I'm thinking that UUID is probably too much here -- thoughts? I feel like we could get away with something like ETL has, a 5-char random generation.

Reply:
Proooobably? But I don't think UUID is too much overhead, and it makes me feel fuzzy and safe. I dig it.

Reply:
I'm with Niko!